File: csharp/main/Implementation/InterfaceNode.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary.Implementation
{
	internal class InterfaceNode : CompositeTypeNode<InterfacePropertyNode>
	{
		public InterfaceNode(String name, List<String> comments, List<String> supertypes)
			: base(name, comments)
		{
			Methods = new List<InterfaceMethodNode>();
			Supertypes = supertypes;
		}

		public List<InterfaceMethodNode> Methods { get; }

		public List<String> Supertypes { get; }
	}
}

File: java/src/main/java/deltix/luminary/implementation/InterfaceNode.java
package deltix.luminary.implementation;

import java.util.ArrayList;
import java.util.List;

public class InterfaceNode extends TypeContainerNode {
    private final String name;
    private final List<DecoratorValueNode> decorators;
    private final List<String> comments;
    private final List<InterfacePropertyNode> properties;
    private final List<InterfaceMethodNode> methods;
    private final List<String> supertypes;

    protected InterfaceNode(String name, List<String> supertypes, List<String> comments) {
        super();
        this.name = name;
        this.comments = comments;
        this.decorators = new ArrayList<>();
        this.properties = new ArrayList<>();
        this.methods = new ArrayList<>();
        this.supertypes = supertypes;
    }

    public List<InterfacePropertyNode> getProperties() {
        return properties;
    }

    public List<InterfaceMethodNode> getMethods() {
        return methods;
    }

    public String getName() {
        return name;
    }

    public List<DecoratorValueNode> getDecorators() {
        return decorators;
    }

    public List<String> getComments() {
        return comments;
    }

    public List<String> getSupertypes() {
        return supertypes;
    }
}
